Pashkevich Anton

prague15031939 opened the pull request:
iterations 1-4 are completed

prague15031939 committed:
initialized cli parser

provided all details of cli interface required by technical task

AlexeiBuzuma commented on final_task/rss_reader.py:
```diff
@@ -0,0 +1,13 @@
+import argparse
+import requests
+
+parser = argparse.ArgumentParser(description='Pure Python command-line RSS reader.', prog='rss_reader')
```
В глобальной секции не должно быть логики. Советую перенести парсинг аргументов в отдельную функцию.

prague15031939 committed:
1 iteration completed

provided work logic for parsingrss feed and displaying news, --json --verbose and --limit optional arguments

prague15031939 committed:
2 iteration completed

cli utility 'rss_reader' was wrapped into distribution package

prague15031939 committed:
updated README.md

prague15031939 committed:
3 iteration completed

provided caching news via database

prague15031939 committed:
fixed news duplication problem

updated README.md in order to illustrate database structure

prague15031939 committed:
fixed table in README.md

AlexeiBuzuma commented on final_task/setup.py:
```diff
@@ -0,0 +1,22 @@
+with open("README.md", "r") as f:
+    long_description = f.read()
+
+setuptools.setup(
```
Согласно тех заданию, должна экспортироваться утилита rss-reader 
после установки пакета через `pip install .` такой утилиты не появляется 
```bash
# rss-reader
bash: rss-reader: command not found
```

AlexeiBuzuma commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,201 @@
+
+    return json.dumps({'news': news}, indent=2)
+
+def main():
```
Возможно стоит разбить эту функцию на несколько маленьких функций

AlexeiBuzuma commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,201 @@
+    """
+    this function prints news in stdout
+    """
+    if len(news) == 0:
```
```python
if not news:
    return None
```

AlexeiBuzuma commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,201 @@
+        try:
+            logger.info(f"checking date..")
+            if not cacher.is_valid_date(args.date):
+                raise ValueError
```
есть смысл добавить текстовое описание к каждому исключению, которое тут вызывается

AlexeiBuzuma commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,201 @@
+    if args.limit:
+        if args.limit < 1:
+            if not args.verbose:
+                print("error: invalid limit value")
```
А зачем для печати одного и того же текста использовать и функцию print и логгер? 
По идее можно обойтись только логгером

prague15031939 committed:
4th iteration completed

provided converting to html and pdf, fixed issues with package distribution

prague15031939 committed:
fixed module import error

provided downloading font neccessary for pdf format from the Internet

prague15031939 committed:
fixed small issues with html displaying

prague15031939 committed:
improved logging mechanics

AlexeiBuzuma commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,223 @@
+            format_converter.to_html(news, args.html)
+        except (OSError, FileNotFoundError):
+            if len(logger.handlers) == 1:
+                logger.addHandler(logging.StreamHandler(sys.stdout))
```
Все время говорит `error: invalid directory`

prague15031939 replied:
@AlexeiBuzuma я перепроверил, всё вроде как нормально

AlexeiBuzuma commented on final_task/rss_reader/format_converter.py:
```diff
@@ -0,0 +1,169 @@
+            while text[i] != '\n':
+                i += 1
+            text = text[:i] + "<br>" + text[i + 1:]
+            i += 4
```
Очень много магических чисел :)

AlexeiBuzuma commented on final_task/rss_reader/format_converter.py:
```diff
@@ -0,0 +1,169 @@
+                break
+            f.write(block)
+
+def to_pdf(news, filepath):
```
Это достаточно большая функция. Есть смысл разделить ее на несколько маленьких

AlexeiBuzuma commented on final_task/rss_reader/format_converter.py:
```diff
@@ -0,0 +1,169 @@
+import requests
+from fpdf import FPDF
+
+def break_lines(text):
```
Возможно я ошибаюсь, но может ли в данном случае подойти метод `replace` у строки?
